packages/components/src/dimension-control/index.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Icon from '../icon';
+import SelectControl from '../select-control';
+import sizesTable, { findSizeBySlug } from './sizes';
+import type { DimensionControlProps, Size } from './types';
+import type { SelectControlProps } from '../select-control/types';
+
+/**
+ * `DimensionControl` is a component designed to provide a UI to control spacing and/or dimensions.
+ *
+ * This feature is still experimental. “Experimental” means this is an early implementation subject to drastic and breaking changes.
+ *
+ * ```jsx
```

Comment:
When visiting the docs tab in Storybook, the code snippet shows twice: once from the JSDoc snippet, once because Storybook generates a dymanic one.

![Screenshot 2023-03-02 at 13 51 00](https://user-images.githubusercontent.com/1083581/222433680-26f26970-51ae-47bb-ab50-6b680e6237e3.jpg)

We should be able to hide the JSDoc snippet by adding the `@example` tag in the JSDoc here:

```suggestion
 * @example
 * ```jsx
```

Reply:
If I remember correctly adding the JSDoc example(s) so that it in fact shows up on the storybook docs page was requested by @mirka (not on this PR). I think the Storybook one (the one with the dark background in your screenshot) is based on the `Default` story. This seems to be the same as with other components. Do we not want to show the JSDoc example on those docs pages?

Reply:
Sometimes the generated snipped from Storybook does not look great because of some limitations in how Storybook generates it, in some other cases it looks fine.

In this case, the generated snippet looks quite clear, and therefore it may make the JSDoc snippet redundant.

(@mirka feel free to add more context if necessary, in case I'm missing something here)

Reply:
We generally have been including the JSDoc snippet in the Storybook. There are definitely pros/cons to it, for example the JSDoc snippet will always include import statements, so it's a more complete snippet ready for copy & paste. On the other hand, it's sometimes redundant and pushes the actual component playground below the fold.

So there are probably ways that we can improve this experience, but for the time being we've always been showing the JSDoc snippet in Storybook, for consistency if nothing else.

Reply:
Sounds good! We can leave the double-snippet as-is :)